lib/instrument/transformer/nskeyed.js

```diff
@@ -202,12 +210,108 @@ class XCTCapabilities extends BaseArchiveHandler {
+  decodeArchive() {
+    return null;
+  }
+}
+
 class NSNull extends BaseArchiveHandler {
+  constructor() {
+    super();
+    this.additionClasses = [NS_OBJECT];
+  }
   decodeArchive () {
     return null;
   }
 }
 
+class XCTestConfiguration extends BaseArchiveHandler {
+  static _default = {
+    aggregateStatisticsBeforeCrash: {
+      XCSuiteRecordsKey: {}
+    },
+    automationFrameworkPath: '/Developer/Library/PrivateFrameworks/XCTAutomationSupport.framework',
+    baselineFileRelativePath: undefined,
+    baselineFileURL: undefined,
+    defaultTestExecutionTimeAllowance: undefined,
+    disablePerformanceMetrics: false,
+    emitOSLogs: false,
+    formatVersion: 2,
+    gatherLocalizableStringsData: false,
+    initializeForUITesting: true,
+    maximumTestExecutionTimeAllowance: undefined,
+    productModuleName: 'WebDriverAgentRunner',
+    randomExecutionOrderingSeed: undefined,
+    reportActivities: true,
+    reportResultsToIDE: true,
+    systemAttachmentLifetime: 2,
+    targetApplicationArguments: [],
+    targetApplicationBundleID: undefined,
+    targetApplicationEnvironment: undefined,
+    targetApplicationPath: '/KEEP-THIS-NOT-EMPTY/KEEP-THIS-NOT-EMPTY',
+    testApplicationDependencies: {},
+    testApplicationUserOverrides: undefined,
+    testBundleRelativePath: undefined,
+    testExecutionOrdering: 0,
+    testTimeoutsEnabled: false,
+    testsDrivenByIDE: false,
+    testsMustRunOnMainThread: true,
+    testsToRun: undefined,
+    testsToSkip: undefined,
+    treatMissingBaselinesAsFailures: false,
+    userAttachmentLifetime: 1
+  }
+  /**
+   *
+   * @param {object} data
+   * @param {string | NSURL} data.testBundleURL
+   * @param {string | NSUUID} data.sessionIdentifier
+   * @param {string?} data.targetApplicationBundleID
+   */
+  constructor(data) {
+    super();
+    if (!data.testBundleURL) {
+      throw new TypeError('testBundleURL cannot be empty');
+    }
+    if (!data.sessionIdentifier) {
+      throw new TypeError('sessionIdentifier cannot be empty');
+    }
+    if (typeof data.testBundleURL === 'string') {
+      data.testBundleURL = new NSURL(undefined, data.testBundleURL);
+    }
+    if (!(data.testBundleURL instanceof NSURL)) {
+      throw new TypeError('Expected testBundleURL to be a valid NSURL instance, got ${typeof testBundleURL} instead');
+    }
+    if (typeof data.sessionIdentifier === 'string') {
+      data.sessionIdentifier = new NSUUID(data.sessionIdentifier);
+    }
+    if (!(data.sessionIdentifier instanceof NSUUID)) {
+      throw new TypeError('Expected sessionIdentifier to be a valid NSUUID instance, got ${typeof testBundleURL} instead');
+    }
+    this._data = { ...XCTestConfiguration._default, ...data };
+  }
+
+  getBytes() {
+    return archive(this);
+  }
+}
+
+class XCActivityRecord extends BaseArchiveHandler {
+  static _keys = ['activityType', 'attachments', 'finish', 'start', 'title', 'uuid'];
+  decodeArchive(archive) {
+    const ret = {};
```

Comment:
this could be simplified to 
```
return XCActivityRecord._keys.reduce((acc, key) => {
  acc[key] = archive.decode(key); 
  return acc;
}, {});
```